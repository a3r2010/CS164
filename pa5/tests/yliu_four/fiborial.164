def fibo = lambda (val) {
	if(val <= 2){
		1
	} else{
		def x = fibo(val-1) + fibo(val-2)
		x
	}
}

def fact = lambda (z) {
	def f = 0
	if (z == 1) {
		f = 1
	} else {
		f = fact(z - 1)
	}
	def a = z * f
	a
}

def fiborial = lambda (f1, f2, V) {
	def x = f1(V) + f2(V)
	def y = f1(V) - f2(V)
	def z = f1(V) * f2(V)
	def alpha = f1(V) / f2(V)
	print x
	print y
	print z
	print alpha
}

def iter = lambda(z) {
	def i = 0;
	lambda () {
		if (i < z){
			i = i + 1
			i
		}
		else {
			null
		}
	}
}

for (x in iter(5)) {
	fiborial(fibo, fact, x)
}
for (y in iter(5)) {
	fiborial(fact, fibo, y)
}