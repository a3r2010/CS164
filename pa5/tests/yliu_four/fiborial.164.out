2
0
1
1
3
-1
2
0
8
-4
12
0
27
-21
72
0
125
-115
600
0
2
0
1
1
3
1
2
2
8
4
12
3
27
21
72
8
125
115
600
24
